tests/xdg_toplevel_stable.cpp

```diff
@@ -222,6 +312,46 @@ TEST_F(XdgToplevelStableTest, interactive_move)
     client.roundtrip();
 }
 
+TEST_F(XdgToplevelStableTest, pointer_leaves_surface_during_interactive_resize)
+{
+    int window_x = 100, window_y = 100;
+    int window_width = 420, window_height = 390;
+    int start_x = window_x + 5, start_y = window_y + 5;
+
+    wlcs::Client client{the_server()};
+    wlcs::Surface surface{client};
+    wlcs::XdgSurfaceStable xdg_shell_surface{client, surface};
+    wlcs::XdgToplevelStable toplevel{xdg_shell_surface};
+    surface.attach_buffer(window_width, window_height);
+    wl_surface_commit(surface);
+    client.roundtrip();
+
+    the_server().move_surface_to(surface, window_x, window_y);
+
+    auto pointer = the_server().create_pointer();
+
+    bool button_down{false};
+    uint32_t last_serial{0};
+
+    client.add_pointer_button_notification([&](uint32_t serial, uint32_t, bool is_down) -> bool {
+            last_serial = serial;
+            button_down = is_down;
+            return true;
+        });
+
+    pointer.move_to(start_x, start_y);
+    pointer.left_button_down();
+
+    client.dispatch_until([&](){
+            return button_down;
+        });
+
+    xdg_toplevel_resize(toplevel, client.seat(), last_serial, XDG_TOPLEVEL_RESIZE_EDGE_TOP_LEFT);
+    client.dispatch_until([&](){
+            return !button_down;
```

Comment:
Likewise, it'd be nice to have the “pointer leaves surface” test gated on the pointer_leave event, rather than the button state.